src/shared/blockdev-util.c

```diff
@@ -94,7 +94,7 @@ int block_device_get_whole_disk(sd_device *dev, sd_device **ret) {
         return 0;
 }
 
-int block_device_get_originating(sd_device *dev, sd_device **ret) {
+int block_device_get_originating(sd_device *dev, bool lookup_originating, sd_device **ret) {
```

Comment:
please do not modify this function. If recursive resolution is needed just call it in a loop at callsite, just like what core/cgroup has already been doing